### ISSUE TITLE
[Package Request] SeaMonkey

### ISSUE THREAD

**Issue by ghost** [void-linux/void-packages#17772]
all-in-one internet application suite.

[http://www.seamonkey-project.org/](url)

While i am aware that this request will either be rejected and closed or put on the back burner, I'd at least want to see my chances of this getting picked up by someone. As Seamonkey is hard to find in any distros repositories and I do believe it offers features other browsers in void's repository currently do not have. Such as having integrated mail, calendar, IRC, etc.

**Comment by q66** [comment]
I'd put this on hold at very least until they update to something that's not based on ancient FF52 codebase. While the added value compared to FF in this case (unlike other firefox forks/variants) might be high enough to make it worth packaging, I also find it potentially bad to ship security issues and so on, especially with the state of modern web nowadays

**Comment by rubyFeedback** [comment]
I agree only partially with q66.

I agree in the sense that mozilla has lost it completely some years ago, so I would be sceptical of their code (actually including FF, so I am not sure why FF should be treated differently to seamonkey here; I'd be wary of anything mozilla touches these days).

HOWEVER had, I disagree on this part:

> make it worth packaging

IMO that should not be up to you, or anyone else to decide, for OTHERS whether their use case includes wanting to use seamonkey or not. If someone wants to use seamonkey, why should you or anyone else forbid this? It's fine if you can not do so or do not want to do so; lack of time is a completely valid reasons. I just disagree that it should be up to others to dictate use cases onto others when it comes to any general purpose linux distribution.

It's perfectly fine to tell people that there are security issues (and this is the case for other projects too, by the way, so I don't see why this arbitrary distinction should exist). You are not responsible for their own actions. That can be handled in a generic manner by labeling packages as "secure" or "not secure" etc...

Building seamonkey, as it is with everything else Mozilla dishes out, is hugely annoying though. I can understand people not being extremely eager to go through it:

http://www.linuxfromscratch.org/blfs/view/cvs/xsoft/seamonkey.html

I typically need either python-2, and/or autoconf-2.13. What madness is this? This madness comes from a code base that has not been maintained in many years really. Other projects transitioned to cmake or meson/ninja and they, for the most part, work fine. Mozilla on the other hand has given up many years ago, and it shows.

I tried to compile the latest /seamonkey-2.49.5 and it fails at this point:

rm -f libmozsandbox.so

/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu/_virtualenv/bin/python /Depot/jjj/seamonkey-2.49.5/mozilla/config/expandlibs_exec.py --uselist --  /usr/bin/g++ -std=gnu++11  -Wall -Wc++11-compat -Wempty-body -Wignored-qualifiers -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wtype-limits -Wunreachable-code -Wwrite-strings -Wno-invalid-offsetof -Wc++14-compat -Wc++1z-compat -Wimplicit-fallthrough -Wno-error=maybe-uninitialized -Wno-error=deprecated-declarations -Wno-error=array-bounds -fno-lifetime-dse -fPIC -fno-exceptions -fno-strict-aliasing -fno-rtti -fno-exceptions -fno-math-errno -pthread -pipe  -g -O2 -fomit-frame-pointer  -fPIC -shared -Wl,-z,defs -Wl,-h,libmozsandbox.so -o libmozsandbox.so  logging.o at_exit.o callback_internal.o lazy_instance.o ref_counted.o singleton.o safe_sprintf.o string16.o string_piece.o string_util.o string_util_constants.o stringprintf.o utf_string_conversion_utils.o utf_string_conversions.o condition_variable_posix.o lock.o lock_impl_posix.o waitable_event_posix.o icu_utf.o platform_thread_internal_posix.o platform_thread_linux.o platform_thread_posix.o thread_collision_warner.o thread_id_name_manager.o thread_local_posix.o thread_restrictions.o time.o time_posix.o bpf_dsl.o codegen.o dump_bpf.o policy.o policy_compiler.o syscall_set.o die.o syscall.o trap.o syscall_wrappers.o LinuxCapabilities.o Sandbox.o SandboxBrokerClient.o SandboxChroot.o SandboxFilter.o SandboxFilterUtil.o SandboxHooks.o SandboxInfo.o SandboxLogging.o SandboxUtil.o SandboxBrokerCommon.o   -lpthread  -Wl,-z,noexecstack -Wl,-z,text -Wl,--build-id -B /Depot/jjj/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu/build/unix/gold    -Wl,-rpath-link,/Depot/jjj/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu/dist/bin -Wl,-rpath-link,/home/Programs/Seamonkey/2.49.5/lib         -ldl  -lrt 
chmod +x libmozsandbox.so
strip  libmozsandbox.so
../../../config/nsinstall -R -m 644 'libmozsandbox.so' '../../../dist/bin'
make[4]: Leaving directory '/Depot/jjj/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu/security/sandbox/linux'
make[3]: Leaving directory '/Depot/jjj/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu'
make[2]: *** [/Depot/jjj/seamonkey-2.49.5/mozilla/config/recurse.mk:33: compile] Error 2
make[2]: Leaving directory '/Depot/jjj/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu'
make[1]: *** [/Depot/jjj/seamonkey-2.49.5/mozilla/config/rules.mk:523: default] Error 2
make[1]: Leaving directory '/Depot/jjj/seamonkey-2.49.5/obj-x86_64-pc-linux-gnu'
make: *** [client.mk:397: build] Error 2

Mozilla is dead IMO. Perhaps others have more luck though; I am using an outdated glibc.

**Comment by silvernode** [comment]
Not to take this discussion off track too much but, in my opinion Mozilla has appeared to start caring a little more in the last year. I think the way they are treating Firefox and the web is honorable when you consider what they could be doing to counteract competitors. As for the code, it's my understanding that they are slowly switching the C++ code base with Rust which is at least a good step in the right direction I think. Maybe my view is completely naive but I like to think Mozilla is beginning to clean up their act and proving they want to continue to serve the web as an open standard.

**Comment by userabc123xyz** [comment]
100% support this request. SeaMonkey is nice.
> Mozilla is dead IMO. Perhaps others have more luck though; I am using an outdated glibc.
SeaMonkey has very little to do with Mozilla, it's an unrelated, autonomous community project.

**Comment by silvernode** [comment]
I did not know that seamonkey was on it's own, that's neat.

**Comment by silvernode** [comment]
> I'd put this on hold at very least until they update to something that's not based on ancient FF52 codebase. While the added value compared to FF in this case (unlike other firefox forks/variants) might be high enough to make it worth packaging, I also find it potentially bad to ship security issues and so on, especially with the state of modern web nowadays

I agree 100%, we should wait on this one.

**Comment by q66** [comment]
Yeah, I have nothing about eventually having this included, but not while it's on an old codebase. Maintaining a web browser is a massive task, and I assume seamonkey does not have a lot of developers to spare, and zero-days are a thing even in mainline Firefox. While some people may claim that distro should not care and it's the user's responsibility, ultimately it's distro maintainers who are blamed when something bad happens, and it's the distro's responsibility to curate what is included in the repository and provide some degree of quality control. Seamonkey has previously announced that they'd be rebasing onto a modern core.

**Comment by userabc123xyz** [comment]
SeaMonkey 2.53 has been released: https://www.seamonkey-project.org/releases/#2.53.1

**Comment by userabc123xyz** [comment]
SeaMonkey 2.53.2 is out: https://www.seamonkey-project.org/releases/#2.53.2

**Comment by userabc123xyz** [comment]
> Seamonkey has previously announced that they'd be rebasing onto a modern core.
It has already moved to a new codebase.

**Comment by xianwenchen** [comment]
It'd be nice to include SeaMonkey.

**Comment by rogerxxxx** [comment]
One of the best basic no frills Internet browsers on the market.

I'll grab the binary built package and install myself.  Chrome is to ad prone, while Firefox screwed their interface.  A default install of Seamonkey, Seamonkey is still the best of both worlds.

Likely in the process of converting to Void, and is my browser of choice for ~20 years or more.
